components/server/src/model/source.py

```diff
@@ -0,0 +1,49 @@
+"""Source model class."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
```

Comment:
This is an interesting solution... Up until now I used quotes if a real import wasn't possible due to circular imports

Reply:
Yeah, must love the __future__ :-)